designs/SearchCommandSpec.md

```diff
@@ -0,0 +1,98 @@
+Microsoft.Extensions.Logging.Abstractions.NullLogger
+
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Microsoft.Extensions.Logging | v5.0.0-preview.6.20305.6 | DLs: 243186566
+Logging infrastructure default implementation for Microsoft.Extensions.Logging.
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Microsoft.IdentityModel.Logging | v6.6.0 | DLs: 120177247
+Includes Event Source based logging support.
+
+--------------------
+> Microsoft.Extensions.Logging.Configuration | v5.0.0-preview.6.20305.6 | DLs: 63205816
+Configuration support for Microsoft.Extensions.Logging.
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Microsoft.Extensions.Logging.Console | v5.0.0-preview.6.20305.6 | DLs: 61796431
+Console logger provider implementation for Microsoft.Extensions.Logging.
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Microsoft.Extensions.Logging.Debug | v5.0.0-preview.6.20305.6 | DLs: 58643995
+Debug output logger provider implementation for Microsoft.Extensions.Logging. This logger logs messages to a debugger monitor by writing messages with System.Diagnostics.Debug.WriteLine().
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Microsoft.Extensions.Logging.EventSource | v5.0.0-preview.6.20305.6 | DLs: 32121683
+EventSource/EventListener logger provider implementation for Microsoft.Extensions.Logging.
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Serilog.Extensions.Logging | v3.0.2-dev-10280 | DLs: 43115462
+-- More  --
+```
+
+## Future Work
+
+* Enabling cross-platform functionality for the search feature
+* Adding optional arguments that allow users to customize their search queries
+
+## Open Questions
+
+* What represents the most effective design choice for displaying the search results, in terms of spacing, demarcation, truncating package descriptions, and so on?
+
+## Considerations
+
+### 1. Combining the results from different sources
+
+Initially, we considered consolidating all search results from the various sources and displaying them together, rather than separating them based on their source. However, users may wish to know where a particular package is located. Also, combining the search results for different sources leads to the complex problem of how to combine these results most effectively.
+
+### 2. Displaying results by printing the output stream to the console without pagination
+
+If we were to simply print the output to the console as a regular stream, the most visible results would be the ones at the end of the output stream, since these would be located right above the next command prompt in the terminal. Pagination, as proposed in this document, ensures that the most relevant results are shown to the users first.
```

Comment:
We've left pagination to 'Future Work' for now because of a lack of consensus on it. To help with the problem of too many results potentially being returned, I have also added an optional argument for -Take, which limits the number of results to display.